### 상관관계 확인

In [41]:
from IPython.core.interactiveshell import InteractiveShell
InteractiveShell.ast_node_interactivity="all"

In [2]:
import numpy as np
import pandas as pd

df = pd.read_csv('./kaggle/input/spaceship-titanic/train_update.csv')
df01 = pd.read_csv('./kaggle/input/spaceship-titanic/train.csv')

In [3]:
df_died = df[df['Transported']==0]

In [44]:
# # Cabin_port
# df_died.groupby('Cabin_port').count()
# df.groupby('Cabin_port').count()
# # HomePlanet
# df_died.groupby('HomePlanet').count()
# df.groupby('HomePlanet').count()
# # CryoSleep
# df_died.groupby('CryoSleep').count()
# df.groupby('CryoSleep').count()
# # Cabin
df_died.groupby('Cabin').count()
df.groupby('Cabin').count()

len(df_died.groupby('Cabin'))

PassengerId  HomePlanet  CryoSleep  Cabin_num  Cabin_port  Destination  \
Cabin                                                                           
0              129         129        129        129         129          129   
1              207         207        207        207         207          207   
2              239         239        239        239         239          239   
3              282         282        282        282         282          282   
4              630         630        630        630         630          630   
5             1583        1583       1583       1583        1583         1583   
6             1240        1240       1240       1240        1240         1240   
7                5           5          5          5           5            5   

        Age   VIP  RoomService  FoodCourt  ShoppingMall   Spa  VRDeck  Name  \
Cabin                                                                         
0       129   129          129        129           129   129     129   125   
1       207   207          207        207           207   207     207   203   
2       239   239          239        239           239   239     239   234   
3       282   282          282        282           282   282     282   278   
4       630   630          630        630           630   630     630   614   
5      1583  1583         1583       1583          1583  1583    1583  1541   
6      1240  1240         1240       1240          1240  1240    1240  1217   
7         5     5            5          5             5     5       5     4   

       Transported  
Cabin               
0              129  
1              207  
2              239  
3              282  
4              630  
5             1583  
6             1240  
7                5

PassengerId  HomePlanet  CryoSleep  Cabin_num  Cabin_port  Destination  \
Cabin                                                                           
0              256         256        256        256         256          256   
1              779         779        779        779         779          779   
2              747         747        747        747         747          747   
3              495         495        495        495         495          495   
4             1025        1025       1025       1025        1025         1025   
5             2821        2821       2821       2821        2821         2821   
6             2564        2564       2564       2564        2564         2564   
7                6           6          6          6           6            6   

        Age   VIP  RoomService  FoodCourt  ShoppingMall   Spa  VRDeck  Name  \
Cabin                                                                         
0       256   256          256        256           256   256     256   249   
1       779   779          779        779           779   779     779   763   
2       747   747          747        747           747   747     747   733   
3       495   495          495        495           495   495     495   485   
4      1025  1025         1025       1025          1025  1025    1025  1005   
5      2821  2821         2821       2821          2821  2821    2821  2747   
6      2564  2564         2564       2564          2564  2564    2564  2506   
7         6     6            6          6             6     6       6     5   

       Transported  
Cabin               
0              256  
1              779  
2              747  
3              495  
4             1025  
5             2821  
6             2564  
7                6

8

In [54]:
df_died.groupby('Cabin_num').count()['PassengerId'][354]

3

In [5]:
# 비율 확인

def died_persent(column_name):
    
    for i in range(len(df_died.groupby(column_name))):
        try:
            df_d0 = df_died.groupby(column_name).count()['PassengerId'][i] 
            df_0 = df.groupby(column_name).count()['PassengerId'][i]  
            print(f'{column_name} {i} : 전체 {df_0} 중 {df_d0}명 {round(df_d0/df_0*100,2)}% 실종')
        except:
            pass

died_persent('Cabin_num')

Cabin_num 0 : 전체 22 중 14명 63.64% 실종
Cabin_num 1 : 전체 15 중 5명 33.33% 실종
Cabin_num 2 : 전체 11 중 6명 54.55% 실종
Cabin_num 3 : 전체 16 중 5명 31.25% 실종
Cabin_num 4 : 전체 7 중 5명 71.43% 실종
Cabin_num 5 : 전체 13 중 6명 46.15% 실종
Cabin_num 6 : 전체 12 중 8명 66.67% 실종
Cabin_num 7 : 전체 9 중 3명 33.33% 실종
Cabin_num 8 : 전체 13 중 7명 53.85% 실종
Cabin_num 9 : 전체 16 중 4명 25.0% 실종
Cabin_num 10 : 전체 14 중 8명 57.14% 실종
Cabin_num 11 : 전체 19 중 8명 42.11% 실종
Cabin_num 12 : 전체 5 중 2명 40.0% 실종
Cabin_num 13 : 전체 16 중 13명 81.25% 실종
Cabin_num 14 : 전체 17 중 5명 29.41% 실종
Cabin_num 15 : 전체 10 중 3명 30.0% 실종
Cabin_num 16 : 전체 17 중 3명 17.65% 실종
Cabin_num 17 : 전체 8 중 6명 75.0% 실종
Cabin_num 18 : 전체 18 중 6명 33.33% 실종
Cabin_num 19 : 전체 22 중 9명 40.91% 실종
Cabin_num 20 : 전체 15 중 9명 60.0% 실종
Cabin_num 21 : 전체 27 중 17명 62.96% 실종
Cabin_num 22 : 전체 15 중 11명 73.33% 실종
Cabin_num 23 : 전체 16 중 10명 62.5% 실종
Cabin_num 24 : 전체 16 중 7명 43.75% 실종
Cabin_num 25 : 전체 15 중 6명 40.0% 실종
Cabin_num 26 : 전체 10 중 1명 10.0% 실종
Cabin_num 27 : 전체 17 중 9명 52.94% 실종
Cabin_num

Cabin_num 232 : 전체 9 중 4명 44.44% 실종
Cabin_num 233 : 전체 15 중 10명 66.67% 실종
Cabin_num 234 : 전체 7 중 4명 57.14% 실종
Cabin_num 235 : 전체 11 중 4명 36.36% 실종
Cabin_num 236 : 전체 8 중 1명 12.5% 실종
Cabin_num 237 : 전체 13 중 5명 38.46% 실종
Cabin_num 238 : 전체 15 중 9명 60.0% 실종
Cabin_num 239 : 전체 11 중 5명 45.45% 실종
Cabin_num 240 : 전체 13 중 8명 61.54% 실종
Cabin_num 241 : 전체 9 중 5명 55.56% 실종
Cabin_num 242 : 전체 9 중 3명 33.33% 실종
Cabin_num 243 : 전체 7 중 4명 57.14% 실종
Cabin_num 244 : 전체 8 중 4명 50.0% 실종
Cabin_num 245 : 전체 11 중 2명 18.18% 실종
Cabin_num 246 : 전체 13 중 10명 76.92% 실종
Cabin_num 247 : 전체 16 중 4명 25.0% 실종
Cabin_num 248 : 전체 8 중 3명 37.5% 실종
Cabin_num 249 : 전체 10 중 1명 10.0% 실종
Cabin_num 250 : 전체 12 중 3명 25.0% 실종
Cabin_num 251 : 전체 10 중 5명 50.0% 실종
Cabin_num 252 : 전체 11 중 4명 36.36% 실종
Cabin_num 253 : 전체 12 중 5명 41.67% 실종
Cabin_num 254 : 전체 9 중 5명 55.56% 실종
Cabin_num 255 : 전체 13 중 5명 38.46% 실종
Cabin_num 256 : 전체 7 중 2명 28.57% 실종
Cabin_num 257 : 전체 9 중 6명 66.67% 실종
Cabin_num 258 : 전체 16 중 6명 37.5% 실종
Cabin_num 259 : 전체 

Cabin_num 493 : 전체 3 중 2명 66.67% 실종
Cabin_num 495 : 전체 2 중 2명 100.0% 실종
Cabin_num 496 : 전체 4 중 4명 100.0% 실종
Cabin_num 497 : 전체 2 중 2명 100.0% 실종
Cabin_num 498 : 전체 3 중 3명 100.0% 실종
Cabin_num 499 : 전체 8 중 4명 50.0% 실종
Cabin_num 500 : 전체 6 중 3명 50.0% 실종
Cabin_num 501 : 전체 3 중 2명 66.67% 실종
Cabin_num 502 : 전체 4 중 2명 50.0% 실종
Cabin_num 504 : 전체 4 중 3명 75.0% 실종
Cabin_num 505 : 전체 8 중 5명 62.5% 실종
Cabin_num 506 : 전체 5 중 4명 80.0% 실종
Cabin_num 508 : 전체 6 중 2명 33.33% 실종
Cabin_num 510 : 전체 5 중 5명 100.0% 실종
Cabin_num 511 : 전체 4 중 2명 50.0% 실종
Cabin_num 512 : 전체 3 중 3명 100.0% 실종
Cabin_num 513 : 전체 6 중 4명 66.67% 실종
Cabin_num 514 : 전체 4 중 3명 75.0% 실종
Cabin_num 515 : 전체 3 중 2명 66.67% 실종
Cabin_num 516 : 전체 6 중 4명 66.67% 실종
Cabin_num 517 : 전체 5 중 1명 20.0% 실종
Cabin_num 519 : 전체 2 중 1명 50.0% 실종
Cabin_num 520 : 전체 3 중 2명 66.67% 실종
Cabin_num 521 : 전체 6 중 6명 100.0% 실종
Cabin_num 522 : 전체 7 중 4명 57.14% 실종
Cabin_num 523 : 전체 7 중 3명 42.86% 실종
Cabin_num 524 : 전체 4 중 3명 75.0% 실종
Cabin_num 525 : 전체 12 중 3명 25.0% 실종
Cab

Cabin_num 775 : 전체 2 중 2명 100.0% 실종
Cabin_num 776 : 전체 3 중 2명 66.67% 실종
Cabin_num 777 : 전체 4 중 2명 50.0% 실종
Cabin_num 778 : 전체 5 중 3명 60.0% 실종
Cabin_num 780 : 전체 3 중 2명 66.67% 실종
Cabin_num 782 : 전체 4 중 3명 75.0% 실종
Cabin_num 783 : 전체 4 중 2명 50.0% 실종
Cabin_num 784 : 전체 3 중 3명 100.0% 실종
Cabin_num 785 : 전체 3 중 1명 33.33% 실종
Cabin_num 786 : 전체 4 중 1명 25.0% 실종
Cabin_num 787 : 전체 3 중 2명 66.67% 실종
Cabin_num 789 : 전체 3 중 1명 33.33% 실종
Cabin_num 790 : 전체 3 중 1명 33.33% 실종
Cabin_num 791 : 전체 3 중 1명 33.33% 실종
Cabin_num 793 : 전체 2 중 2명 100.0% 실종
Cabin_num 794 : 전체 4 중 2명 50.0% 실종
Cabin_num 795 : 전체 3 중 1명 33.33% 실종
Cabin_num 796 : 전체 3 중 1명 33.33% 실종
Cabin_num 797 : 전체 3 중 2명 66.67% 실종
Cabin_num 799 : 전체 5 중 1명 20.0% 실종
Cabin_num 800 : 전체 2 중 2명 100.0% 실종
Cabin_num 801 : 전체 2 중 1명 50.0% 실종
Cabin_num 803 : 전체 3 중 1명 33.33% 실종
Cabin_num 804 : 전체 3 중 2명 66.67% 실종
Cabin_num 805 : 전체 5 중 3명 60.0% 실종
Cabin_num 806 : 전체 4 중 2명 50.0% 실종
Cabin_num 807 : 전체 2 중 1명 50.0% 실종
Cabin_num 808 : 전체 4 중 1명 25.0% 실종
Cabi

Cabin_num 1083 : 전체 4 중 1명 25.0% 실종
Cabin_num 1085 : 전체 2 중 1명 50.0% 실종
Cabin_num 1086 : 전체 4 중 1명 25.0% 실종
Cabin_num 1087 : 전체 6 중 1명 16.67% 실종
Cabin_num 1088 : 전체 5 중 3명 60.0% 실종
Cabin_num 1089 : 전체 3 중 1명 33.33% 실종
Cabin_num 1094 : 전체 3 중 1명 33.33% 실종
Cabin_num 1095 : 전체 4 중 1명 25.0% 실종
Cabin_num 1096 : 전체 2 중 1명 50.0% 실종
Cabin_num 1098 : 전체 5 중 1명 20.0% 실종
Cabin_num 1099 : 전체 3 중 3명 100.0% 실종
Cabin_num 1100 : 전체 4 중 1명 25.0% 실종
Cabin_num 1102 : 전체 4 중 2명 50.0% 실종
Cabin_num 1103 : 전체 1 중 1명 100.0% 실종
Cabin_num 1104 : 전체 4 중 3명 75.0% 실종
Cabin_num 1105 : 전체 3 중 1명 33.33% 실종
Cabin_num 1106 : 전체 3 중 1명 33.33% 실종
Cabin_num 1107 : 전체 1 중 1명 100.0% 실종
Cabin_num 1108 : 전체 8 중 3명 37.5% 실종
Cabin_num 1109 : 전체 6 중 5명 83.33% 실종
Cabin_num 1111 : 전체 4 중 2명 50.0% 실종
Cabin_num 1112 : 전체 3 중 2명 66.67% 실종
Cabin_num 1113 : 전체 4 중 3명 75.0% 실종
Cabin_num 1114 : 전체 5 중 1명 20.0% 실종
Cabin_num 1115 : 전체 3 중 2명 66.67% 실종
Cabin_num 1116 : 전체 2 중 1명 50.0% 실종
Cabin_num 1117 : 전체 3 중 1명 33.33% 실종
Cabin_num 1118 :

Cabin_num 1382 : 전체 4 중 3명 75.0% 실종
Cabin_num 1383 : 전체 4 중 2명 50.0% 실종
Cabin_num 1384 : 전체 2 중 2명 100.0% 실종
Cabin_num 1385 : 전체 2 중 1명 50.0% 실종
Cabin_num 1386 : 전체 2 중 2명 100.0% 실종
Cabin_num 1387 : 전체 5 중 1명 20.0% 실종
Cabin_num 1388 : 전체 2 중 1명 50.0% 실종
Cabin_num 1390 : 전체 2 중 1명 50.0% 실종
Cabin_num 1391 : 전체 3 중 1명 33.33% 실종
Cabin_num 1392 : 전체 4 중 2명 50.0% 실종
Cabin_num 1393 : 전체 2 중 1명 50.0% 실종
Cabin_num 1395 : 전체 3 중 3명 100.0% 실종
Cabin_num 1396 : 전체 5 중 4명 80.0% 실종
Cabin_num 1397 : 전체 3 중 2명 66.67% 실종
Cabin_num 1398 : 전체 3 중 1명 33.33% 실종
Cabin_num 1399 : 전체 3 중 3명 100.0% 실종
Cabin_num 1400 : 전체 3 중 1명 33.33% 실종
Cabin_num 1402 : 전체 4 중 2명 50.0% 실종
Cabin_num 1403 : 전체 1 중 1명 100.0% 실종
Cabin_num 1405 : 전체 2 중 1명 50.0% 실종
Cabin_num 1406 : 전체 2 중 2명 100.0% 실종
Cabin_num 1407 : 전체 1 중 1명 100.0% 실종
Cabin_num 1408 : 전체 3 중 3명 100.0% 실종
Cabin_num 1409 : 전체 4 중 3명 75.0% 실종
Cabin_num 1410 : 전체 4 중 2명 50.0% 실종
Cabin_num 1411 : 전체 12 중 7명 58.33% 실종
Cabin_num 1412 : 전체 3 중 1명 33.33% 실종
Cabin_num 141

In [38]:
# # 비율 확인

# def died_persent(column_name):
#     df_d0 =df_died.groupby(column_name).count()['PassengerId'][0] #
#     df_d1 = df_died.groupby(column_name).count()['PassengerId'][1] #
#     df_0 = df.groupby(column_name).count()['PassengerId'][0] # 
#     df_1 = df.groupby(column_name).count()['PassengerId'][1] #
#     print(f'{column_name} 0 : 전체 {df_0} 중 {df_d0}명 {round(df_d0/df_0*100,2)}% 실종')
#     print(f'{column_name} 1 : 전체 {df_1} 중 {df_d1}명 {round(df_d1/df_1*100,2)}% 실종')

# died_persent('CryoSleep')

CryoSleep 0 : 전체 5568 중 3741명 67.19% 실종
CryoSleep 1 : 전체 3125 중 574명 18.37% 실종


In [29]:
574/3125*100

18.368000000000002

### 도착하지 못한 사람
Transported == 0
1. Cabin_port
    - 좌현(P,0) 전체 4403 중 2405 명 54.6% 실종 
    - 우현(S,0) 전체 4290 중 1910 명 44.5% 실종
2. HomePlanet
    - 비슷한 비율로 실종된것 처럼 보인다 
    - Earth:0 : 전체 4714 중 2719 명 57.7% 실종 
    - Mars:1 : 전체 2175 중 740 명 34.0% 실종 
    - Europa:2 : 전체 1804 중 856 명 47.5% 실종 
3. CryoSleep(방 밖으로 나가지 않는 사람)
    - CryoSleep 0 : 전체 5568 중 3741명 67.19% 실종
    - CryoSleep 1 : 전체 3125 중 574명 18.37% 실종
4. Cabin
    - Cabin 0 : 전체 256 중 129명 50.39% 실종
    - Cabin 1 : 전체 779 중 207명 26.57% 실종
    - Cabin 2 : 전체 747 중 239명 31.99% 실종
    - Cabin 3 : 전체 495 중 282명 56.97% 실종
    - Cabin 4 : 전체 1025 중 630명 61.46% 실종
    - Cabin 5 : 전체 2821 중 1583명 56.11% 실종
    - Cabin 6 : 전체 2564 중 1240명 48.36% 실종
    - Cabin 7 : 전체 6 중 5명 83.33% 실종
    

모든 퍼센트는 소수 2째자리 반올림

In [7]:
def died_persent(column_name):
    
    for i in range(len(df_died.groupby(column_name))):
        try:
            df_d0 = df_died.groupby(column_name).count()['PassengerId'][i] 
            df_0 = df.groupby(column_name).count()['PassengerId'][i]  
            print(f'{column_name} {i} : 전체 {df_0} 중 {df_d0}명 {round(df_d0/df_0*100,2)}% 실종')
        except:
            pass

died_persent('Age')

Age 0 : 전체 178 중 34명 19.1% 실종
Age 1 : 전체 67 중 18명 26.87% 실종
Age 2 : 전체 75 중 22명 29.33% 실종
Age 3 : 전체 75 중 16명 21.33% 실종
Age 4 : 전체 71 중 18명 25.35% 실종
Age 5 : 전체 34 중 14명 41.18% 실종
Age 6 : 전체 40 중 17명 42.5% 실종
Age 7 : 전체 53 중 21명 39.62% 실종
Age 8 : 전체 50 중 21명 42.0% 실종
Age 9 : 전체 44 중 20명 45.45% 실종
Age 10 : 전체 49 중 21명 42.86% 실종
Age 11 : 전체 49 중 21명 42.86% 실종
Age 12 : 전체 50 중 20명 40.0% 실종
Age 13 : 전체 146 중 73명 50.0% 실종
Age 14 : 전체 156 중 74명 47.44% 실종
Age 15 : 전체 163 중 71명 43.56% 실종
Age 16 : 전체 154 중 73명 47.4% 실종
Age 17 : 전체 166 중 73명 43.98% 실종
Age 18 : 전체 363 중 178명 49.04% 실종
Age 19 : 전체 317 중 161명 50.79% 실종
Age 20 : 전체 284 중 153명 53.87% 실종
Age 21 : 전체 328 중 188명 57.32% 실종
Age 22 : 전체 294 중 159명 54.08% 실종
Age 23 : 전체 292 중 155명 53.08% 실종
Age 24 : 전체 324 중 184명 56.79% 실종
Age 25 : 전체 243 중 126명 51.85% 실종
Age 26 : 전체 268 중 140명 52.24% 실종
Age 27 : 전체 260 중 133명 51.15% 실종
Age 28 : 전체 267 중 117명 43.82% 실종
Age 29 : 전체 231 중 118명 51.08% 실종
Age 30 : 전체 186 중 101명 54.3% 실종
Age 31 : 전체 202 중 119명 5

In [12]:
def died_persent(column_name):
    
    for i in range(len(df_died.groupby(column_name))):
        try:
            df_d0 = df_died.groupby(column_name).count()['PassengerId'][i] 
            df_0 = df.groupby(column_name).count()['PassengerId'][i]  
            # print(f'{column_name} {i} : 전체 {df_0} 중 {df_d0}명 {round(df_d0/df_0*100,2)}% 실종')
            p_died = round(df_d0/df_0*100,2)
            if p_died >= 80:
                print(f'{column_name} {i} : 전체 {df_0} 중 {df_d0}명 {round(df_d0/df_0*100,2)}% 실종')
                
        except:
            pass

died_persent('Age')

Age 69 : 전체 6 중 5명 83.33% 실종
Age 73 : 전체 7 중 6명 85.71% 실종
Age 79 : 전체 3 중 3명 100.0% 실종


In [13]:
def died_persent(column_name):
    
    for i in range(len(df_died.groupby(column_name))):
        try:
            df_d0 = df_died.groupby(column_name).count()['PassengerId'][i] 
            df_0 = df.groupby(column_name).count()['PassengerId'][i]  
            print(f'{column_name} {i} : 전체 {df_0} 중 {df_d0}명 {round(df_d0/df_0*100,2)}% 실종')
            p_died = round(df_d0/df_0*100,2)
            #if p_died >= 80:
            #    print(f'{column_name} {i} : 전체 {df_0} 중 {df_d0}명 {round(df_d0/df_0*100,2)}% 실종')
                
        except:
            pass

died_persent('VIP')

VIP 0 : 전체 8494 중 4192명 49.35% 실종
VIP 1 : 전체 199 중 123명 61.81% 실종


In [21]:
df_sortFood = df.sort_values(by=['FoodCourt'])
df_sortFood

PassengerId  HomePlanet  CryoSleep  Cabin  Cabin_num  Cabin_port  \
0        0001_01           1          0      1          0           0   
5058     6893_01           1          1      1        261           1   
5057     6891_01           2          0      3        215           0   
5055     6883_01           0          0      6       1120           1   
5054     6882_02           2          0      4        438           0   
...          ...         ...        ...    ...        ...         ...   
7635     1969_01           1          0      2         75           1   
2721     3796_01           1          0      1        121           0   
2459     3444_01           1          0      1        131           1   
927      1292_01           1          0      1         55           1   
1590     2215_01           1          0      0         27           1   

      Destination  Age  VIP  RoomService  FoodCourt  ShoppingMall   Spa  \
0               2   39    0            0          0             0     0   
5058            2   29    0            0          0             0     0   
5057            2   27    0          725          0          1660     0   
5055            2   26    0            0          0             1     0   
5054            2   53    0            0          0             0     0   
...           ...  ...  ...          ...        ...           ...   ...   
7635            0   27    0         2160      21066             0  2448   
2721            0   33    0           90      26830             0    27   
2459            0   33    0            0      27071             0    15   
927             0   31    1            0      27723             3   202   
1590            0   41    1            0      29813             0  1257   

      VRDeck                Name  Transported  
0          0     Maham Ofracculy            0  
5058       0   Winork Dreamspric            1  
5057       0        Treynx Casty            1  
5055     398  Evaned Wolferguson            1  
5054       0         Giamon Dake            0  
...      ...                 ...          ...  
7635    2293      Unukath Sandul            1  
2721     703     Dhena Semageary            1  
2459       0    Ainoxa Scerodbox            1  
927     3146      Scharab Conale            1  
1590       6  Pulchib Quidedbolt            1  

[8693 rows x 16 columns]

In [25]:
food_positive = df_sortFood[df_sortFood['FoodCourt']>0]
food_positive

PassengerId  HomePlanet  CryoSleep  Cabin  Cabin_num  Cabin_port  \
3837     5269_01           0          0      4        336           1   
1614     2251_02           1          0      0         29           1   
3552     4898_01           0          0      5        933           1   
3874     5331_01           0          0      5       1095           0   
3501     4807_01           0          0      6        776           0   
...          ...         ...        ...    ...        ...         ...   
7635     1969_01           1          0      2         75           1   
2721     3796_01           1          0      1        121           0   
2459     3444_01           1          0      1        131           1   
927      1292_01           1          0      1         55           1   
1590     2215_01           1          0      0         27           1   

      Destination  Age  VIP  RoomService  FoodCourt  ShoppingMall   Spa  \
3837            2   21    0           38          1           591     1   
1614            2   58    0            2          1            14     2   
3552            2   28    0          647          1            32    71   
3874            2   23    0            0          1           292     0   
3501            2   44    0           22          1             0   864   
...           ...  ...  ...          ...        ...           ...   ...   
7635            0   27    0         2160      21066             0  2448   
2721            0   33    0           90      26830             0    27   
2459            0   33    0            0      27071             0    15   
927             0   31    1            0      27723             3   202   
1590            0   41    1            0      29813             0  1257   

      VRDeck                Name  Transported  
3837      31   Mollen Ewisentley            1  
1614    1153   Zaurion Hoodjuded            0  
3552     140   Lorian Bertsontry            1  
3874     340  Coreen Woodgezalez            0  
3501       0        Elinda Miley            0  
...      ...                 ...          ...  
7635    2293      Unukath Sandul            1  
2721     703     Dhena Semageary            1  
2459       0    Ainoxa Scerodbox            1  
927     3146      Scharab Conale            1  
1590       6  Pulchib Quidedbolt            1  

[3222 rows x 16 columns]

In [29]:
food_positive.describe()

HomePlanet    CryoSleep        Cabin    Cabin_num   Cabin_port  \
count  3222.000000  3222.000000  3222.000000  3222.000000  3222.000000   
mean      0.549348     0.017381     3.861887   537.634389     0.495345   
std       0.664824     0.130705     1.826156   514.074760     0.500056   
min       0.000000     0.000000     0.000000     0.000000     0.000000   
25%       0.000000     0.000000     2.000000   132.000000     0.000000   
50%       0.000000     0.000000     5.000000   300.500000     0.000000   
75%       1.000000     0.000000     5.000000   894.000000     1.000000   
max       2.000000     1.000000     7.000000  1894.000000     1.000000   

       Destination          Age          VIP   RoomService     FoodCourt  \
count  3222.000000  3222.000000  3222.000000   3222.000000   3222.000000   
mean      1.476723    31.926133     0.040968    316.337678   1240.317505   
std       0.836113    12.895400     0.198248    818.164166   2429.683831   
min       0.000000     0.000000     0.000000      0.000000      1.000000   
25%       1.000000    22.000000     0.000000      0.000000     45.000000   
50%       2.000000    29.000000     0.000000      1.000000    413.000000   
75%       2.000000    40.000000     0.000000    216.750000   1185.500000   
max       2.000000    79.000000     1.000000  14327.000000  29813.000000   

       ShoppingMall           Spa        VRDeck  Transported  
count   3222.000000   3222.000000   3222.000000  3222.000000  
mean     244.169770    722.265984    720.984171     0.353507  
std      818.165023   1745.971587   1766.406211     0.478133  
min        0.000000      0.000000      0.000000     0.000000  
25%        0.000000      0.000000      0.000000     0.000000  
50%        1.000000     39.500000     41.500000     0.000000  
75%      143.750000    632.500000    605.750000     1.000000  
max    23492.000000  22408.000000  24133.000000     1.000000

In [51]:
food_positive['FoodCourt'].quantile(.25)

45.0

In [33]:
### FoodCourt 에서 3사분위수 이상의 금액을 지불한 사람들 ###
### 즉, FoodCourt에서 전체 금액 분포 중 75퍼센트 이상의 금액을 지불한 사람들 ###

food_75per = food_positive[food_positive['FoodCourt']>=1185]
food_75per
food_75per.groupby('Transported').count()

PassengerId  HomePlanet  CryoSleep  Cabin  Cabin_num  Cabin_port  \
Transported                                                                     
0                    354         354        354    354        354         354   
1                    452         452        452    452        452         452   

             Destination  Age  VIP  RoomService  FoodCourt  ShoppingMall  Spa  \
Transported                                                                     
0                    354  354  354          354        354           354  354   
1                    452  452  452          452        452           452  452   

             VRDeck  Name  
Transported                
0               354   343  
1               452   442

In [38]:
food_75per.iloc[:,[0,10,15]].groupby('Transported').count()

PassengerId  FoodCourt
Transported                        
0                    354        354
1                    452        452

In [42]:
def per_count(per):
    df_sortFood = df.sort_values(by=['FoodCourt'])
    food_positive = df_sortFood[df_sortFood['FoodCourt']>0]

    if per == 25:
        food_per = food_positive[food_positive['FoodCourt']>=45]
        print(food_per.iloc[:,[10,15]].groupby('Transported').count())
    elif per == 50:
        food_per = food_positive[food_positive['FoodCourt']>=413]
        print(food_per.iloc[:,[10,15]].groupby('Transported').count())
    elif per == 75:
        food_per = food_positive[food_positive['FoodCourt']>=1185]
        print(food_per.iloc[:,[10,15]].groupby('Transported').count())
    else:
        pass

per_count(25)
per_count(50)
per_count(75)


             FoodCourt
Transported           
0                 1442
1                  977
             FoodCourt
Transported           
0                  849
1                  762
             FoodCourt
Transported           
0                  354
1                  452


In [85]:
def per_count(per, col_name):
    
    df_sort = df.sort_values(by=[col_name])
    df_positive = df_sort[df_sort[col_name]>0]
    
    df_per = df_positive[df_positive[col_name]>= df_positive[col_name].quantile(per)]
    
    if col_name == 'FoodCourt':
        count = df_per.iloc[:,[10,15]].groupby('Transported').count()
        print(count)
        print("Transported 실패 비율 ",count.iloc[0,:]/(count.iloc[0,:] + count.iloc[1,:]))
    elif col_name == 'RoomService':
        count = df_per.iloc[:,[9,15]].groupby('Transported').count()
        print(count)
        print("Transported 실패 비율 ",count.iloc[0,:]/(count.iloc[0,:] + count.iloc[1,:]))
    elif col_name == 'ShoppingMall':
        count = df_per.iloc[:,[11,15]].groupby('Transported').count()
        print(count)
        print("Transported 실패 비율 ",count.iloc[0,:]/(count.iloc[0,:] + count.iloc[1,:]))
    elif col_name == 'Spa':
        count = df_per.iloc[:,[12,15]].groupby('Transported').count()
        print(count)
        print("Transported 실패 비율 ",count.iloc[0,:]/(count.iloc[0,:] + count.iloc[1,:]))


In [86]:
    
per_count(.25, 'FoodCourt')
per_count(.50, 'FoodCourt')
per_count(.60, 'FoodCourt')
per_count(.75, 'FoodCourt')


             FoodCourt
Transported           
0                 1442
1                  977
Transported 실패 비율  FoodCourt    0.596114
dtype: float64
             FoodCourt
Transported           
0                  849
1                  762
Transported 실패 비율  FoodCourt    0.527002
dtype: float64
             FoodCourt
Transported           
0                  646
1                  644
Transported 실패 비율  FoodCourt    0.500775
dtype: float64
             FoodCourt
Transported           
0                  354
1                  452
Transported 실패 비율  FoodCourt    0.439206
dtype: float64


In [87]:
    
per_count(.25, 'RoomService')
per_count(.50, 'RoomService')
per_count(.60, 'RoomService')
per_count(.75, 'RoomService')


             RoomService
Transported             
0                   1782
1                    559
Transported 실패 비율  RoomService    0.761213
dtype: float64
             RoomService
Transported             
0                   1266
1                    292
Transported 실패 비율  RoomService    0.81258
dtype: float64
             RoomService
Transported             
0                   1035
1                    213
Transported 실패 비율  RoomService    0.829327
dtype: float64
             RoomService
Transported             
0                    664
1                    115
Transported 실패 비율  RoomService    0.852375
dtype: float64


In [88]:
    
per_count(.25, 'ShoppingMall')
per_count(.50, 'ShoppingMall')
per_count(.60, 'ShoppingMall')
per_count(.75, 'ShoppingMall')


             ShoppingMall
Transported              
0                    1478
1                     779
Transported 실패 비율  ShoppingMall    0.654852
dtype: float64
             ShoppingMall
Transported              
0                     885
1                     620
Transported 실패 비율  ShoppingMall    0.58804
dtype: float64
             ShoppingMall
Transported              
0                     661
1                     543
Transported 실패 비율  ShoppingMall    0.549003
dtype: float64
             ShoppingMall
Transported              
0                     367
1                     388
Transported 실패 비율  ShoppingMall    0.486093
dtype: float64


In [89]:
    
per_count(.25, 'Spa')
per_count(.50, 'Spa')
per_count(.60, 'Spa')
per_count(.75, 'Spa')


              Spa
Transported      
0            1880
1             647
Transported 실패 비율  Spa    0.743965
dtype: float64
              Spa
Transported      
0            1341
1             344
Transported 실패 비율  Spa    0.795846
dtype: float64
              Spa
Transported      
0            1090
1             258
Transported 실패 비율  Spa    0.808605
dtype: float64
             Spa
Transported     
0            680
1            163
Transported 실패 비율  Spa    0.806643
dtype: float64


In [44]:
food_zero = df_sortFood[df_sortFood['FoodCourt']==0]
food_zero

PassengerId  HomePlanet  CryoSleep  Cabin  Cabin_num  Cabin_port  \
0        0001_01           1          0      1          0           0   
5058     6893_01           1          1      1        261           1   
5057     6891_01           2          0      3        215           0   
5055     6883_01           0          0      6       1120           1   
5054     6882_02           2          0      4        438           0   
...          ...         ...        ...    ...        ...         ...   
2572     3588_01           2          0      3        120           0   
2768     3868_06           0          1      6        630           1   
2573     3588_02           2          0      3        120           0   
2574     3591_01           0          0      4        214           0   
2563     3569_01           2          1      5        741           0   

      Destination  Age  VIP  RoomService  FoodCourt  ShoppingMall  Spa  \
0               2   39    0            0          0             0    0   
5058            2   29    0            0          0             0    0   
5057            2   27    0          725          0          1660    0   
5055            2   26    0            0          0             1    0   
5054            2   53    0            0          0             0    0   
...           ...  ...  ...          ...        ...           ...  ...   
2572            2   35    0          576          0           459   22   
2768            2   42    0            0          0             0    0   
2573            2   40    0          787          0            88    0   
2574            2   22    0            0          0             0    0   
2563            2   19    0            0          0             0    0   

      VRDeck                Name  Transported  
0          0     Maham Ofracculy            0  
5058       0   Winork Dreamspric            1  
5057       0        Treynx Casty            1  
5055     398  Evaned Wolferguson            1  
5054       0         Giamon Dake            0  
...      ...                 ...          ...  
2572       0        Frunad Concy            0  
2768       0        Line Braymon            1  
2573       0         Terry Concy            0  
2574    1609  Elenay Serraynardy            0  
2563       0           Prah Bine            1  

[5471 rows x 16 columns]

In [45]:
food_zero.iloc[:,[10,15]].groupby('Transported').count()

FoodCourt
Transported           
0                 2232
1                 3239

In [50]:
food_positive.iloc[:,[8,15]].groupby('Transported').count()

VIP
Transported      
0            2083
1            1139

In [73]:
food_positive[food_positive['FoodCourt'] <= food_positive['FoodCourt'].quantile(.75)].iloc[:,[10,15]].groupby('Transported').count()

FoodCourt
Transported           
0                 1729
1                  687

In [70]:
food_positive[food_positive['FoodCourt'] >= food_positive['FoodCourt'].quantile(.90)].iloc[:,[8,10]].groupby('VIP').count()

FoodCourt
VIP           
0          291
1           32

### 저체온증 관련

선장이 식당을 제외한 모든 곳에서 난방과 온수 시스템을 끄라고 지시 -> FoodCourt 와 실종과의 관련성
[해당 링크](https://namu.wiki/w/%ED%83%80%EC%9D%B4%ED%83%80%EB%8B%89%ED%98%B8%20%EC%B9%A8%EB%AA%B0%20%EC%82%AC%EA%B3%A0)